cmd/loki-canary/main.go

```diff
@@ -52,10 +61,30 @@ func main() {
 	sentChan := make(chan time.Time)
 	receivedChan := make(chan time.Time)
 
-	w := writer.NewWriter(os.Stdout, sentChan, *interval, *size)
-	r := reader.NewReader(os.Stderr, receivedChan, *tls, *addr, *user, *pass, *lName, *lVal)
-	c := comparator.NewComparator(os.Stderr, *wait, *pruneInterval, *buckets, sentChan, receivedChan, r, true)
+	c := &canary{}
```

Comment:
you could make start a function of c too no ? like stop ?

Reply:
I was considering it but decided against it b/c it would have to take like 10 parameters.  This seemed less gross.